nvflare/app_common/workflows/statistics_controller.py

```diff
@@ -0,0 +1,286 @@
+        self.metrics_task_flow(abort_signal, fl_ctx, clients, StC.STATS_1st_METRICS)
+        self.metrics_task_flow(abort_signal, fl_ctx, clients, StC.STATS_2nd_METRICS)
+        self.post_fn(self.task_name, fl_ctx)
+
+        self.log_info(fl_ctx, f"task {self.task_name} control flow end.")
+
+    def start_controller(self, fl_ctx: FLContext):
+        pass
+
+    def stop_controller(self, fl_ctx: FLContext):
+        pass
+
+    def process_result_of_unknown_task(
+        self, client: Client, task_name: str, client_task_id: str, result: Shareable, fl_ctx: FLContext
+    ):
+        pass
+
+    def metrics_task_flow(self, abort_signal: Signal, fl_ctx: FLContext, clients: List[Client], metric_task: str):
+
+        self.log_info(fl_ctx, f"start prepare inputs for task {metric_task}")
+        inputs = self._prepare_inputs(metric_task, fl_ctx)
+        results_cb_fn = self._get_result_cb(metric_task)
+
+        self.log_info(fl_ctx, f"task: {self.task_name} metrics_flow for {metric_task} started.")
+
+        if abort_signal.triggered:
+            return False
+
+        task = Task(name=self.task_name, data=inputs, result_received_cb=results_cb_fn)
+
+        self.broadcast_and_wait(
+            task=task,
+            targets=None,
+            min_responses=len(clients),
+            fl_ctx=fl_ctx,
+            wait_time_after_min_received=1,
+            abort_signal=abort_signal,
+        )
+
+        self.global_metrics = get_global_stats(self.global_metrics, self.client_metrics, metric_task)
+
+        self.log_info(fl_ctx, f"task {self.task_name} metrics_flow for {metric_task} flow end.")
+
+    def results_cb(self, client_task: ClientTask, fl_ctx: FLContext):
+        client_name = client_task.client.name
+        task_name = client_task.task.name
+        self.log_info(fl_ctx, f"Processing {task_name} result from client {client_name}")
+
+        result = client_task.result
+        rc = result.get_return_code()
+        print("return code =", rc)
+        if rc == ReturnCode.OK:
+            self.log_info(fl_ctx, f"Received result entries from client:{client_name}, " f"for task {task_name}")
+            dxo = from_shareable(result)
+            client_result = dxo.data
+
+            metric_task = client_result[StC.METRIC_TASK_KEY]
+            self.log_info(fl_ctx, f"handle client {client_name} results for metrics task: {metric_task}")
+            metrics = client_result[metric_task]
+            for metric in metrics:
+                if metric not in self.client_metrics:
+                    self.client_metrics[metric] = {client_name: metrics[metric]}
+                else:
+                    self.client_metrics[metric].update({client_name: metrics[metric]})
+
+            if metric_task == StC.STATS_1st_METRICS:
+                ds_features = client_result[StC.STATS_FEATURES]
+                if ds_features:
+                    self.client_features.update({client_name: ds_features})
```

Comment:
Do we need to check the metric task?
could just be
```
ds_features = client_result.get(StC.STATS_FEATURES)
if ds_features:
    self.client_features.update({client_name: ds_features})
    
```

Reply:
If the client return features, the controller should update.
This check "if metric_task == StC.STATS_1st_METRICS" is not NECESSARY.
But I am fine keep this check.

suggest use `ds_features = client_result.get(StC.STATS_FEATURES)`